packages/trpc/server/routers/viewer/organizations/addBulkTeams.handler.ts

```diff
@@ -0,0 +1,86 @@
+import { isOrganisationAdmin } from "@calcom/lib/server/queries/organisations";
+import { prisma } from "@calcom/prisma";
+import type { Prisma } from "@calcom/prisma/client";
+import { MembershipRole } from "@calcom/prisma/enums";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TrpcSessionUser } from "../../../trpc";
+import type { TAddBulkTeams } from "./addBulkTeams.schema";
+
+type AddBulkTeamsHandler = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TAddBulkTeams;
+};
+
+export async function addBulkTeamsHandler({ ctx, input }: AddBulkTeamsHandler) {
+  const currentUser = ctx.user;
+
+  if (!currentUser.organizationId) throw new TRPCError({ code: "UNAUTHORIZED" });
+
+  // check if user is admin of organization
+  if (!(await isOrganisationAdmin(currentUser?.id, currentUser.organizationId)))
```

Comment:
Might be worth creating a trpc procedure to check this on the router

Reply:
Maybe - this is a NIT tho and can be refactored along with the rest of org routes that require admin.